src/routes/file-storage-api/accessValidations.ts

```diff
@@ -1,28 +1,98 @@
+} from '@icgc-argo/ego-token-utils';
+import { EGO_DACO_POLICY_NAME } from 'config';
 
-export const hasSufficientProgramMembershipAccess = async (config: {
+export const hasSufficientProgramMembershipAccess = (config: {
   scopes: PermissionScopeObj[];
   file?: EsFileCentricDocument;
-}): Promise<boolean> => {
-  /**
-   * @todo: actually implement this function
-   */
+}): boolean => {
   if (config.scopes && config.file) {
-    return true;
-  } else {
-    return true;
+    const { scopes, file } = config;
+    const releaseStage = file.release_stage;
+
+    const serializedScopes = scopes.map(egoTokenUtils.serializeScope);
+    const accessLevel = egoTokenUtils.getProgramMembershipAccessLevel({
+      permissions: serializedScopes,
+    });
+
+    const programId = file.study_id;
+
+    if (releaseStage === FILE_RELEASE_STAGE.OWN_PROGRAM) {
+      return (
+        accessLevel === UserProgramMembershipAccessLevel.DCC_MEMBER ||
+        egoTokenUtils.canReadProgramData({ permissions: serializedScopes, programId })
+      );
+    }
+
+    if (releaseStage === FILE_RELEASE_STAGE.FULL_PROGRAMS) {
+      return (
+        accessLevel === UserProgramMembershipAccessLevel.DCC_MEMBER ||
+        egoTokenUtils.canReadProgramData({ permissions: serializedScopes, programId }) ||
+        accessLevel === UserProgramMembershipAccessLevel.FULL_PROGRAM_MEMBER
+      );
+    }
+
+    if (releaseStage === FILE_RELEASE_STAGE.ASSOCIATE_PROGRAMS) {
+      return (
+        accessLevel === UserProgramMembershipAccessLevel.DCC_MEMBER ||
+        egoTokenUtils.canReadProgramData({ permissions: serializedScopes, programId }) ||
+        accessLevel === UserProgramMembershipAccessLevel.FULL_PROGRAM_MEMBER ||
+        accessLevel === UserProgramMembershipAccessLevel.ASSOCIATE_PROGRAM_MEMBER
+      );
+    }
+
+    if (releaseStage === FILE_RELEASE_STAGE.PUBLIC) {
+      const hasAccess =
+        accessLevel === UserProgramMembershipAccessLevel.DCC_MEMBER ||
+        egoTokenUtils.canReadProgramData({ permissions: serializedScopes, programId }) ||
+        accessLevel === UserProgramMembershipAccessLevel.FULL_PROGRAM_MEMBER ||
+        accessLevel === UserProgramMembershipAccessLevel.ASSOCIATE_PROGRAM_MEMBER ||
+        accessLevel === UserProgramMembershipAccessLevel.PUBLIC_MEMBER;
+
+      return file.file_access === FILE_ACCESS.CONTROLLED
+        ? hasAccess && hasSufficientDacoAccess({ scopes })
+        : hasAccess;
+    }
   }
+  return false;
 };
 
-export const hasSufficientDacoAccess = async (config: {
-  scopes: PermissionScopeObj[];
-  file?: EsFileCentricDocument;
-}): Promise<boolean> => {
-  return true;
+export const hasSufficientDacoAccess = (config: { scopes: PermissionScopeObj[] }): boolean => {
+  const dacoScopes = config.scopes.filter(({ policy }) => policy === EGO_DACO_POLICY_NAME);
+
+  return dacoScopes.length > 0 && dacoScopes.every(scope => scope.permission === PERMISSIONS.READ)
+    ? true
+    : false;
```

Comment:
`dacoScopes.length > 0 && dacoScopes.every(scope => scope.permission === PERMISSIONS.READ)` is already a positive boolean, no need for this extra boolean conversion :)